roboteam_ai/src/control/ControlUtils.cpp

```diff
@@ -235,7 +235,22 @@ double ControlUtils::calculateMaxAcceleration(const Vector2 &vel, double angle)
     double a = abs(toVectorDiff.x);
     return Constants::MAX_ACC_UPPER() * (a) + Constants::MAX_ACC_LOWER() * (1-a);
 }
-
+Vector2 ControlUtils::accelerationLimiterNew(const Vector2 &vel,double robotAngle, const Vector2 &prevVel) {
+    // calculate max
+    Vector2 diff=vel-prevVel;
+    Angle angleDiff=diff.angle()-robotAngle;
+    Vector2 toVectorDiff=angleDiff.toVector2();
+    double a=abs(toVectorDiff.x);
```

Comment:
Why use toVectorDiff, isn't `a=abs(cos(angleDiff))` sufficient?

Reply:
This was exactly the same code as you used previously. It's functionally the same, so I don't mind changing it.